core/modules/layout/includes/layout.class.inc

```diff
@@ -176,7 +176,7 @@ class Layout {
    *
    * @var LayoutContext[]
    */
-  private $contexts;
+  var $contexts = array();
```

Comment:
I think we want to be explicit: private, public or protected.

Reply:
My OOP is poor. Why only this property? None of the other properties are explicit.

Reply:
Took a look. `var` is now a synonym for `public` so I guess that means it's technically not deprecated in PHP. https://stackoverflow.com/questions/1206105/what-does-php-keyword-var-do

In this case, I believe it was explicitly set to `private` because there are `getContexts` and `setContexts` methods on this class so don't want people to access that variable directly.

Reply:
Better approach, in my view, is to update the docblock for `$contexts` so it matches that of `$path` higher up:

```
  /**
   * The internal path of a layout.
   *
   * The layout path must match the provided menu item path. Set and get this
   * variable's value with Layout::getPath() and Layout::setPath().
   *
   * @var string
   */
  private $path;
```

Reply:
This sounds reasonable. I had assumed it was originally private because there was no external need to access contexts and setting was all on the fly (the setter function is new). What is the advantage of a private property though? Tried googling that but too much info.

Reply:
From what I understand, it's to prevent other classes from messing up a variable that should only be used internally to a class. It's also a way to control how to access the variables contents via the setter and getter methods.

Reply:
Basically:
private: Only the class may access, use, and modify properties and methods with this scope.
protected: The class and any clases that extend it may access, use, and modify properties and methods with this scope.
public: Anything may access, use, and modify properties and methods with this scope.

The advantage of a private property is to control who or what may access, use, and modify the property. If it's public, anything can directly change it at any time, which could lead to unpredictable behavior, depending on the class and what it's used for.

Generally, I prefer to scope properties as private and methods as private, protected, or public, depending on who or what should be allowed to access, use, and modify the specific data of the object, and then create setters and getters for directly accessing, using, and modifying the properties.

And, the scope should be specified on the other properties as well, but if they're not set to public, it will likely be an API breaking change.

Reply:
Resolved in latest commits